src/goto-programs/goto_convert.cpp

```diff
@@ -174,17 +311,29 @@ void goto_convertt::finish_gotos(goto_programt &dest, const irep_idt &mode)
         // goto_programt::instructionst is std::list.
       }
 
-      // We don't currently handle variables *entering* scope, which
-      // is illegal for C++ non-pod types and impossible in Java in any case.
-      // This is however valid C.
+      // Variables *entering* scope on goto, is illegal for C++ non-pod types
+      // and impossible in Java. This is however valid C and should be taken
```

Comment:
⛏️ Actually in C it's illegal to jump entering the scope of a variable-length array or variably-modified typed variables, so the comment is wrong and should rephrased.

However in general it is possible, so I'm just raising as a :pick and not blocking

Reply:
I have added this detail to the comment.